edge-util/src/Microsoft.Azure.Devices.Edge.Storage/InMemoryDbStoreProvider.cs

```diff
@@ -1,18 +1,100 @@
 // Copyright (c) Microsoft. All rights reserved.
 namespace Microsoft.Azure.Devices.Edge.Storage
 {
+    using System;
     using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
     using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
 
     public class InMemoryDbStoreProvider : IDbStoreProvider
```

Comment:
As an overall approach, I think we should have a separate class that takes care of BackupAndRestore - 
Takes in a IDictionary or KeyedCollection and backs it up, or reads from the backup and returns a list of Dictionaries or KeyedCollections. 
These can then be used to populate and backup the DB. 
That will allow us to separate the logic, and test/maintain/replace it more easily. 
Note that we should still use decorator pattern - but it will simply call the Backup and Resotre apis on this interface.

Reply:
Sure, I'll take a stab at it and update the PR

Reply:
Updated the PR using the decorator pattern for DbStoreProvider.

Abstracted the backup and restore of both DBStores and the data specifically using the IDbStoreBackupRestore and IBackupRestore interfaces. This way the individual components can be replaced as needed.